File: Java/BestTimeToBuyAndSellStock2_122.java
import java.util.*;
class Solution {
    public int maxProfit(int[] prices) {
        int profit =0;
        for(int i=1;i<prices.length;i++){
            if (prices[i] > prices[i-1]){
                profit+=prices[i]-prices[i-1];
            }
        }
        return profit;
    }
}

public class BestTimeToBuyAndSellStock2_122{

    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        Solution sol=new Solution();
        System.out.print("Stock prices : ");
        String[] input = sc.nextLine().split(" ");
                int[] nums = new int[input.length];
                for (int i = 0; i < input.length; i++) {
                    nums[i] = Integer.parseInt(input[i]);
                }
        
        System.out.printf("Max Profit : %d" , sol.maxProfit(nums));
        }
    }

// Input: prices = [7,1,5,3,6,4]
// Output: 7

// Input: prices = [1,2,3,4,5]
// Output: 4

// Input: prices = [7,6,4,3,1]
// Output: 0

File: LeetCode/Python/BestTimeToBuyAndSellStock2_122.py
from typing import List
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit=0
        for i in range(1,len(prices)):
            if prices[i] > prices[i-1]:
                profit+=prices[i]-prices[i-1]

        return profit
        
s=Solution()
n=int(input("Number of test cases : "))

for i in range(n):
    ls=list(map(int,input("Stock Prices: ").split()))
    print(f"Max Profit : {s.maxProfit(ls)}")
    
    
# Input: prices = [7,1,5,3,6,4]
# Output: 7

# Input: prices = [1,2,3,4,5]
# Output: 4

# Input: prices = [7,6,4,3,1]
# Output: 0